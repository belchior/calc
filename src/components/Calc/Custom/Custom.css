/* Calc */
/*
  Calc - Custom

  Helper for unit convertion
  1px = 0.0625em
  2px = 0.125em
  3px = 0.1875em
  4px = 0.25em
  5px = 0.3125em
  6px = 0.375em
  7px = 0.4375em
  8px = 0.5em
  9px = 0.5625em
  10px = 0.625em
  11px = 0.6875em
  12px = 0.75em
  13px = 0.8125em
  14px = 0.875em
  15px = 0.9375em
  16px = 1em

*/
.Custom {
  box-sizing: border-box;
  display: flex;
  flex-flow: row wrap;
  border-radius: 0.375em;
  max-width: 29em;
  background: linear-gradient(to bottom, #ffffff 0%, #cccccc 5.06125em, #eeeeee 5.06125em, #dddddd 100%) no-repeat border-box, linear-gradient(to bottom, #888888 0%, #444444 5.06125em, #dddddd 5.06125em, #cccccc 100%) no-repeat border-box;
  background-size: 94% 100%, 100% 100%;
  background-position: 50% 0, 0 0;
  background-origin: padding-box, padding-box;
  box-shadow: 0 0 0.1875em rgba(0, 0, 0, 0.5);
}
.Custom .display {
  overflow: hidden;
  box-sizing: border-box;
  margin: 1em 15%;
  border-radius: 0.375em;
  border: 0.0625em solid #bbbbbb;
  padding: 0 1em;
  width: 70%;
  height: 3em;
  line-height: 3em;
  text-align: right;
  color: #666666;
  background-color: #dddddd;
}
.Custom .display:focus {
  border: 0.05em solid #7E5AB8;
  color: #ffffff;
  background-color: #998FB8;
  outline: none;
}
.Custom .keyboard {
  box-sizing: border-box;
  display: flex;
  padding: 0.5em 4.5%;
}
.Custom .functions {
  max-width: 37.67%;
}
.Custom .arithmetic {
  max-width: 62.33%;
}
.Custom .functions,
.Custom .arithmetic {
  box-sizing: border-box;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-content: space-between;
  align-items: center;
  flex-basis: auto;
}
.Custom .btn {
  order: 1;
  margin: 0.3125em;
  border-radius: 0.375em;
  border: 0.0625em solid rgba(255, 255, 255, 0.1);
  width: 2.5em;
  height: 2.5em;
  text-align: center;
  line-height: 2.5em;
  color: #ffffff;
  background-image: linear-gradient(to bottom, #888888, #777777);
  box-shadow: 0 -0.125em 0.125em rgba(255, 255, 255, 0.6), 0 0.0625em 0.125em rgba(0, 0, 0, 0.8);
  cursor: pointer;
  transition: all 0.3s cubic-bezier(0, 0, 0.2, 1);
}
.Custom .btn:hover {
  background-image: linear-gradient(to bottom, #999999, #777777);
}
.Custom .btn.double {
  width: 6em;
}
.Custom .btn[data-name="delete"] {
  background-image: linear-gradient(to bottom, #d85656, #d82c2c);
}
.Custom .btn[data-name="delete"]:hover {
  background-image: linear-gradient(to bottom, #e16060, #df3333);
}
.Custom .btn[data-name="clear"] {
  background-image: linear-gradient(to bottom, #78a3b0, #6995a5);
}
.Custom .btn[data-name="clear"]:hover {
  background-image: linear-gradient(to bottom, #84afb8, #719daa);
}
.Custom .btn[data-name="power"] span {
  position: relative;
  top: -7px;
  left: 2px;
  font-size: 0.5em;
}
.Custom .btn:focus {
  background-image: linear-gradient(to bottom, #8869c5, #7e5ab8);
}
.Custom .btn:focus:hover {
  background-image: linear-gradient(to bottom, #9074cd, #8364c1);
}
@media screen and (max-width: 29em) {
  .Custom {
    max-width: 19em;
  }
  .Custom .display {
    margin: 1em 10%;
    width: 80%;
  }
  .Custom .keyboard {
    flex-flow: column;
  }
  .Custom .functions {
    max-width: 100%;
  }
  .Custom .arithmetic {
    max-width: 100%;
  }
  .functions .btn.double {
    margin: 0.4465em;
    width: 3.5715em;
    height: 3.5715em;
    font-size: 0.7em;
  }
  .Custom .btn[data-name="delete"] {order: 1;}
  .Custom .btn[data-name="pi"] {order: 3;}
  .Custom .btn[data-name="clear"] {order: 2;}
  .Custom .btn[data-name="madd"] {order: 4;}
  .Custom .btn[data-name="msubtract"] {order: 5;}
  .Custom .btn[data-name="power"] {order: 6;}
  .Custom .btn[data-name="sqrt"] {order: 7;}
  .Custom .btn[data-name="percentage"] {order: 8;}
  .Custom .btn[data-name="mrecall"] {order: 9;}
  .Custom .btn[data-name="mclear"] {order: 10;}
}
