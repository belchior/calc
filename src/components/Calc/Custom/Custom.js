import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Custom.css';

class Custom extends Component {
  render() {
    let className = 'Custom';
    if (this.props.error) {
      className += ' shake-horizontal';
      this.props.disableError();
    }
    return (
      <form className={className}>
        <p className="display">{this.props.formula}</p>
        <div className="keyboard">
          <div className="functions">
            <button onClick={this.props.percentageClick} type="button" className="btn" title="percentage" data-name="percentage" data-value="%">%</button>
            <button onClick={this.props.memoryPlusClick} type="button" className="btn" title="memory plus" data-name="madd">m+</button>
            <button onClick={this.props.memoryMinusClick} type="button" className="btn" title="memory minus" data-name="msubtract">m−</button>
            <button onClick={this.props.powerClick} type="button" className="btn" title="power" data-name="power" data-value="^">x<span>Y</span></button>
            <button onClick={this.props.memoryClearClick} type="button" className="btn" title="memory clear" data-name="mclear">mc</button>
            <button onClick={this.props.memoryRecallClick} type="button" className="btn" title="memory recall" data-name="mrecall">mr</button>
            <button onClick={this.props.sqrtClick} type="button" className="btn" title="square root" data-name="sqrt" data-value="√">√</button>
            <button onClick={this.props.deleteClick} type="button" className="btn double" title="delete" data-name="delete">delete</button>
            <button onClick={this.props.piClick} type="button" className="btn" title="constant PI" data-name="pi" data-value="π">π</button>
            <button onClick={this.props.clearClick} type="button" className="btn double" title="clear" data-name="clear">clear</button>
          </div>
          <div className="arithmetic">
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 7" data-name="number7" data-value="7">7</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 8" data-name="number8" data-value="8">8</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 9" data-name="number9" data-value="9">9</button>
            <button onClick={this.props.plusClick} type="button" className="btn" title="plus" data-name="addition" data-value="+">+</button>
            <button onClick={this.props.multiplicationClick} type="button" className="btn" title="times" data-name="multiplication" data-value="×">×</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 4" data-name="number4" data-value="4">4</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 5" data-name="number5" data-value="5">5</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 6" data-name="number6" data-value="6">6</button>
            <button onClick={this.props.minusClick} type="button" className="btn" title="minus" data-name="subtraction" data-value="-">−</button>
            <button onClick={this.props.divisionClick} type="button" className="btn" title="divided" data-name="division" data-value="÷">÷</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 1" data-name="number1" data-value="1">1</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 2" data-name="number2" data-value="2">2</button>
            <button onClick={this.props.numberClick} type="button" className="btn" title="number 3" data-name="number3" data-value="3">3</button>
            <button onClick={this.props.parenthesisLeftClick} type="button" className="btn" title="parenthesis opening" data-name="parenthesisOpen" data-value="(">(</button>
            <button onClick={this.props.parenthesisRightClick} type="button" className="btn" title="parenthesis closing" data-name="parenthesisClose" data-value=")">)</button>
            <button onClick={this.props.numberClick} type="button" className="btn double" title="number 0" data-name="number0" data-value="0">0</button>
            <button onClick={this.props.dotClick} type="button" className="btn" title="dot" data-name="dot" data-value=".">.</button>
            <button onClick={this.props.equalsClick} type="button" className="btn double" title="equals" data-name="equality">=</button>
          </div>
        </div>
      </form>
    );
  }
}

export default Custom;

Custom.propTypes = {
  clearClick: PropTypes.func.isRequired,
  deleteClick: PropTypes.func.isRequired,
  disableError: PropTypes.func.isRequired,
  divisionClick: PropTypes.func.isRequired,
  dotClick: PropTypes.func.isRequired,
  equalsClick: PropTypes.func.isRequired,
  error: PropTypes.bool,
  formula: PropTypes.string,
  memoryPlusClick: PropTypes.func.isRequired,
  memoryMinusClick: PropTypes.func.isRequired,
  memoryClearClick: PropTypes.func.isRequired,
  memoryRecallClick: PropTypes.func.isRequired,
  minusClick: PropTypes.func.isRequired,
  multiplicationClick: PropTypes.func.isRequired,
  numberClick: PropTypes.func.isRequired,
  parenthesisLeftClick: PropTypes.func.isRequired,
  parenthesisRightClick: PropTypes.func.isRequired,
  percentageClick: PropTypes.func.isRequired,
  piClick: PropTypes.func.isRequired,
  plusClick: PropTypes.func.isRequired,
  powerClick: PropTypes.func.isRequired,
  sqrtClick: PropTypes.func.isRequired,
};
