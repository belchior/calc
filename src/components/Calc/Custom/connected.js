import {connect} from 'react-redux';
import Custom from './Custom';

export const mapStateToProps = (state) => {
  return state.custom;
};

export const mapDispatchToProps = (dispatch) => {
  return {
    clearClick(e) {
      dispatch({type: 'CLEAR', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    deleteClick(e) {
      dispatch({type: 'DELETE', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    divisionClick(e) {
      dispatch({type: 'DIVISION', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    dotClick(e) {
      dispatch({type: 'DOT', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    equalsClick(e) {
      dispatch({type: 'EQUALS', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    memoryClearClick(e) {
      dispatch({type: 'MEMORY_CLEAR', calc: 'custom'});
    },
    memoryMinusClick(e) {
      dispatch({type: 'MEMORY_MINUS', calc: 'custom'});
    },
    memoryPlusClick(e) {
      dispatch({type: 'MEMORY_PLUS', calc: 'custom'});
    },
    memoryRecallClick(e) {
      dispatch({type: 'MEMORY_RECALL', calc: 'custom'});
    },
    minusClick(e) {
      dispatch({type: 'MINUS', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    multiplicationClick(e) {
      dispatch({type: 'MULTIPLY', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    numberClick(e) {
      dispatch({type: 'NUMBER', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    parenthesisLeftClick(e) {
      dispatch({type: 'PARENTHESIS_LEFT', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    parenthesisRightClick(e) {
      dispatch({type: 'PARENTHESIS_RIGHT', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    percentageClick(e) {
      dispatch({type: 'PERCENTAGE', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    piClick(e) {
      dispatch({type: 'PI', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    plusClick(e) {
      dispatch({type: 'PLUS', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    powerClick(e) {
      dispatch({type: 'POWER', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    sqrtClick(e) {
      dispatch({type: 'SQRT', calc: 'custom', payload: e.target.getAttribute('data-value')});
    },
    disableError() {
      setTimeout(() => dispatch({type: 'ERROR', calc: 'custom', payload: false}), 400);
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Custom);
