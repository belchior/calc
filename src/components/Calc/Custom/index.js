import Custom, { mapStateToProps, mapDispatchToProps } from './connected';

export { mapStateToProps, mapDispatchToProps, };
export default Custom;
