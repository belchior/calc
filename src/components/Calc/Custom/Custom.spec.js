import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Custom from './Custom';

const setup = (props = {}) => {
  const requiredProps = {
    clearClick: () => {},
    deleteClick: () => {},
    disableError: () => {},
    divisionClick: () => {},
    dotClick: () => {},
    equalsClick: () => {},
    memoryPlusClick: () => {},
    memoryMinusClick: () => {},
    memoryClearClick: () => {},
    memoryRecallClick: () => {},
    minusClick: () => {},
    multiplicationClick: () => {},
    numberClick: () => {},
    parenthesisLeftClick: () => {},
    parenthesisRightClick: () => {},
    percentageClick: () => {},
    piClick: () => {},
    plusClick: () => {},
    powerClick: () => {},
    sqrtClick: () => {},
    ...props,
  };
  return shallow(<Custom {...requiredProps} />);
};

describe('Custom', () => {
  it('should render without crashing', () => {
    const component = () => setup();
    expect(component).not.toThrow();
  });

  it('should add css class shake-horizontal when error props is equal true', () => {
    const props = {
      disableError: jest.fn(),
      error: true,
    };
    const wrapper = setup(props);
    const custom = toJson(wrapper);

    expect(props.disableError).toHaveBeenCalled();
    expect(custom).toMatchSnapshot();
  });

  it('should render the value of formula into element with css class display', () => {
    const props = { formula: '789' };
    const wrapper = setup(props);
    const custom = toJson(wrapper);

    expect(custom).toMatchSnapshot();
  });
});
