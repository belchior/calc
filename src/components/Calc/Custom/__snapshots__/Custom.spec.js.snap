// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom should add css class shake-horizontal when error props is equal true 1`] = `
<form
  className="Custom shake-horizontal"
>
  <p
    className="display"
  />
  <div
    className="keyboard"
  >
    <div
      className="functions"
    >
      <button
        className="btn"
        data-name="percentage"
        data-value="%"
        onClick={[Function]}
        title="percentage"
        type="button"
      >
        %
      </button>
      <button
        className="btn"
        data-name="madd"
        onClick={[Function]}
        title="memory plus"
        type="button"
      >
        m+
      </button>
      <button
        className="btn"
        data-name="msubtract"
        onClick={[Function]}
        title="memory minus"
        type="button"
      >
        m−
      </button>
      <button
        className="btn"
        data-name="power"
        data-value="^"
        onClick={[Function]}
        title="power"
        type="button"
      >
        x
        <span>
          Y
        </span>
      </button>
      <button
        className="btn"
        data-name="mclear"
        onClick={[Function]}
        title="memory clear"
        type="button"
      >
        mc
      </button>
      <button
        className="btn"
        data-name="mrecall"
        onClick={[Function]}
        title="memory recall"
        type="button"
      >
        mr
      </button>
      <button
        className="btn"
        data-name="sqrt"
        data-value="√"
        onClick={[Function]}
        title="square root"
        type="button"
      >
        √
      </button>
      <button
        className="btn double"
        data-name="delete"
        onClick={[Function]}
        title="delete"
        type="button"
      >
        delete
      </button>
      <button
        className="btn"
        data-name="pi"
        data-value="π"
        onClick={[Function]}
        title="constant PI"
        type="button"
      >
        π
      </button>
      <button
        className="btn double"
        data-name="clear"
        onClick={[Function]}
        title="clear"
        type="button"
      >
        clear
      </button>
    </div>
    <div
      className="arithmetic"
    >
      <button
        className="btn"
        data-name="number7"
        data-value="7"
        onClick={[Function]}
        title="number 7"
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        data-name="number8"
        data-value="8"
        onClick={[Function]}
        title="number 8"
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        data-name="number9"
        data-value="9"
        onClick={[Function]}
        title="number 9"
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        data-name="addition"
        data-value="+"
        onClick={[Function]}
        title="plus"
        type="button"
      >
        +
      </button>
      <button
        className="btn"
        data-name="multiplication"
        data-value="×"
        onClick={[Function]}
        title="times"
        type="button"
      >
        ×
      </button>
      <button
        className="btn"
        data-name="number4"
        data-value="4"
        onClick={[Function]}
        title="number 4"
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        data-name="number5"
        data-value="5"
        onClick={[Function]}
        title="number 5"
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        data-name="number6"
        data-value="6"
        onClick={[Function]}
        title="number 6"
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        data-name="subtraction"
        data-value="-"
        onClick={[Function]}
        title="minus"
        type="button"
      >
        −
      </button>
      <button
        className="btn"
        data-name="division"
        data-value="÷"
        onClick={[Function]}
        title="divided"
        type="button"
      >
        ÷
      </button>
      <button
        className="btn"
        data-name="number1"
        data-value="1"
        onClick={[Function]}
        title="number 1"
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        data-name="number2"
        data-value="2"
        onClick={[Function]}
        title="number 2"
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        data-name="number3"
        data-value="3"
        onClick={[Function]}
        title="number 3"
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        data-name="parenthesisOpen"
        data-value="("
        onClick={[Function]}
        title="parenthesis opening"
        type="button"
      >
        (
      </button>
      <button
        className="btn"
        data-name="parenthesisClose"
        data-value=")"
        onClick={[Function]}
        title="parenthesis closing"
        type="button"
      >
        )
      </button>
      <button
        className="btn double"
        data-name="number0"
        data-value="0"
        onClick={[Function]}
        title="number 0"
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        data-name="dot"
        data-value="."
        onClick={[Function]}
        title="dot"
        type="button"
      >
        .
      </button>
      <button
        className="btn double"
        data-name="equality"
        onClick={[Function]}
        title="equals"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</form>
`;

exports[`Custom should render the value of formula into element with css class display 1`] = `
<form
  className="Custom"
>
  <p
    className="display"
  >
    789
  </p>
  <div
    className="keyboard"
  >
    <div
      className="functions"
    >
      <button
        className="btn"
        data-name="percentage"
        data-value="%"
        onClick={[Function]}
        title="percentage"
        type="button"
      >
        %
      </button>
      <button
        className="btn"
        data-name="madd"
        onClick={[Function]}
        title="memory plus"
        type="button"
      >
        m+
      </button>
      <button
        className="btn"
        data-name="msubtract"
        onClick={[Function]}
        title="memory minus"
        type="button"
      >
        m−
      </button>
      <button
        className="btn"
        data-name="power"
        data-value="^"
        onClick={[Function]}
        title="power"
        type="button"
      >
        x
        <span>
          Y
        </span>
      </button>
      <button
        className="btn"
        data-name="mclear"
        onClick={[Function]}
        title="memory clear"
        type="button"
      >
        mc
      </button>
      <button
        className="btn"
        data-name="mrecall"
        onClick={[Function]}
        title="memory recall"
        type="button"
      >
        mr
      </button>
      <button
        className="btn"
        data-name="sqrt"
        data-value="√"
        onClick={[Function]}
        title="square root"
        type="button"
      >
        √
      </button>
      <button
        className="btn double"
        data-name="delete"
        onClick={[Function]}
        title="delete"
        type="button"
      >
        delete
      </button>
      <button
        className="btn"
        data-name="pi"
        data-value="π"
        onClick={[Function]}
        title="constant PI"
        type="button"
      >
        π
      </button>
      <button
        className="btn double"
        data-name="clear"
        onClick={[Function]}
        title="clear"
        type="button"
      >
        clear
      </button>
    </div>
    <div
      className="arithmetic"
    >
      <button
        className="btn"
        data-name="number7"
        data-value="7"
        onClick={[Function]}
        title="number 7"
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        data-name="number8"
        data-value="8"
        onClick={[Function]}
        title="number 8"
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        data-name="number9"
        data-value="9"
        onClick={[Function]}
        title="number 9"
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        data-name="addition"
        data-value="+"
        onClick={[Function]}
        title="plus"
        type="button"
      >
        +
      </button>
      <button
        className="btn"
        data-name="multiplication"
        data-value="×"
        onClick={[Function]}
        title="times"
        type="button"
      >
        ×
      </button>
      <button
        className="btn"
        data-name="number4"
        data-value="4"
        onClick={[Function]}
        title="number 4"
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        data-name="number5"
        data-value="5"
        onClick={[Function]}
        title="number 5"
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        data-name="number6"
        data-value="6"
        onClick={[Function]}
        title="number 6"
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        data-name="subtraction"
        data-value="-"
        onClick={[Function]}
        title="minus"
        type="button"
      >
        −
      </button>
      <button
        className="btn"
        data-name="division"
        data-value="÷"
        onClick={[Function]}
        title="divided"
        type="button"
      >
        ÷
      </button>
      <button
        className="btn"
        data-name="number1"
        data-value="1"
        onClick={[Function]}
        title="number 1"
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        data-name="number2"
        data-value="2"
        onClick={[Function]}
        title="number 2"
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        data-name="number3"
        data-value="3"
        onClick={[Function]}
        title="number 3"
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        data-name="parenthesisOpen"
        data-value="("
        onClick={[Function]}
        title="parenthesis opening"
        type="button"
      >
        (
      </button>
      <button
        className="btn"
        data-name="parenthesisClose"
        data-value=")"
        onClick={[Function]}
        title="parenthesis closing"
        type="button"
      >
        )
      </button>
      <button
        className="btn double"
        data-name="number0"
        data-value="0"
        onClick={[Function]}
        title="number 0"
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        data-name="dot"
        data-value="."
        onClick={[Function]}
        title="dot"
        type="button"
      >
        .
      </button>
      <button
        className="btn double"
        data-name="equality"
        onClick={[Function]}
        title="equals"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</form>
`;
